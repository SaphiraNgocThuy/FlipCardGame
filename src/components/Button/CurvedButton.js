import React, { memo } from "react";
import PropTypes from "prop-types";
import { ButtonText, ButtonContainer } from "./ButtonStyles";

const CurvedButton = ({ title, onPress, left }) => {
  return (
    <ButtonContainer {...{ onPress, left }}>
      <ButtonText>{title}</ButtonText>
    </ButtonContainer>
  );
};

CurvedButton.propTypes = {
  title: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
  left: PropTypes.bool,
};

export default memo(CurvedButton);
