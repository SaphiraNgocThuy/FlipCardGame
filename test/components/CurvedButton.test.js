import React from 'react';
import renderer from 'react-test-renderer';
import CurvedButton from '../../src/components/Button/CurvedButton';

it('render normal button', () => {
  const tree = renderer
    .create(<CurvedButton title="test" onPress={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('render left side button', () => {
  const tree = renderer
    .create(<CurvedButton title="test" onPress={() => {}} left />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
