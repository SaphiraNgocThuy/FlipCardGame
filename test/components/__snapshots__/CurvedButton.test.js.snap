// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render left side button 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#2ac5db",
      "borderBottomRightRadius": 30,
      "borderTopRightRadius": 30,
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 16,
      "paddingLeft": 16,
      "paddingRight": 24,
      "paddingTop": 16,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontSize": 20,
          "fontWeight": "bold",
        },
      ]
    }
  >
    test
  </Text>
</View>
`;

exports[`render normal button 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#2ac5db",
      "borderRadius": 30,
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 16,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 16,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontSize": 20,
          "fontWeight": "bold",
        },
      ]
    }
  >
    test
  </Text>
</View>
`;
